docs/Coding/guidelines.rst

```diff
@@ -261,6 +294,9 @@ Containers in the Standard Library should be preferred over Qt ones:
 One notable exception is ``QStringList``, which should be preferred over other constructs because it
 integrates better with Qt.
 
+The main point here is to avoid using function parameters to return values.
```

Comment:
Not really the main point (which is rather avoiding explicit memory management), but a nice side-effect. Would tone down a bit.

Reply:
If you have a wording suggestion, I'd gladly add it in next commit.

Reply:
Feel like it's a bit specific for this style guide, but here we go:
> This can have consequences on function signatures. For instance, a common C idiom is to pass an output array and a size as function parameter, whereas this is achieved in C++ by returning a `std::vector` (or, if performance is critical, passing a reference to a vector that the function fills).